dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/DefaultEventDataQueryService.java

```diff
@@ -224,7 +224,16 @@ public EventQueryParams getFromRequest( EventDataQueryRequest request )
 
         if ( request.getOutputType() == EventOutputType.ENROLLMENT )
         {
-            params.withStartEndDatesForPeriods( false );
+            if ( request.getStartDate() == null || request.getEndDate() == null )
```

Comment:
You can also use `!request.hasStartEndDate()` for more a expressive syntax.

Reply:
Done